ui/src/app/settings/views/Scripts/ScriptsList.js

```diff
@@ -0,0 +1,211 @@
+import { format, parse } from "date-fns";
+import PropTypes from "prop-types";
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import classNames from "classnames";
+import { Link } from "react-router-dom";
+
+import "./ScriptsList.scss";
+import Button from "app/base/components/Button";
+import Code from "app/base/components/Code";
+import Col from "app/base/components/Col";
+import Loader from "app/base/components/Loader";
+import MainTable from "app/base/components/MainTable";
+import Row from "app/base/components/Row";
+import TableDeleteConfirm from "app/base/components/TableDeleteConfirm";
+import SearchBox from "app/base/components/SearchBox";
+import actions from "app/settings/actions";
+import selectors from "app/settings/selectors";
+
+const generateRows = (
+  scripts,
+  expandedId,
+  setExpandedId,
+  expandedType,
+  setExpandedType,
+  hideExpanded
+) =>
+  scripts.map(script => {
+    const expanded = expandedId === script.id;
+    const showDelete = expandedType === "delete";
+
+    const [lastHistory] = script.history.slice(-1);
+
+    let scriptSrc;
+    if (lastHistory.data) {
+      try {
+        scriptSrc = atob(lastHistory.data);
+      } catch {
+        console.error(`Unable to decode script src for ${script.name}.`);
+      }
+    }
+
+    // history timestamps are in the format: Mon, 02 Sep 2019 02:02:39 -0000
+    let uploadedOn;
+    if (lastHistory && lastHistory.created) {
+      try {
+        uploadedOn = format(
+          parse(
+            lastHistory.created,
+            "EEE, dd LLL yyyy HH:mm:ss xxxx",
+            new Date()
+          ),
+          "yyyy-LL-dd H:mm"
+        );
+      } catch (error) {
+        console.error(`Unable to parse timestamp for ${script.name}.`);
+      }
+    } else {
+      uploadedOn = "Never";
+    }
+
+    return {
+      className: expanded ? "p-table__row is-active" : null,
+      columns: [
+        {
+          content: (
+            <Button
+              appearance="link"
+              className={classNames("scripts-list__toggle", {
+                "is-active": expanded && !showDelete
+              })}
+              inline
+              onClick={() => {
+                if (expandedId && !showDelete) {
+                  hideExpanded();
+                } else {
+                  setExpandedId(script.id);
+                  setExpandedType("details");
+                }
+              }}
+            >
+              <span className="scripts-list__toggle-name">{script.name}</span>
+            </Button>
+          ),
+          role: "rowheader"
+        },
+        {
+          content: script.description
+        },
+        { content: uploadedOn },
+        {
+          content: (
+            <>
+              <span className="p-tooltip p-tooltip--left">
```

Comment:
This tooltip span doesn't actually need to be there, looks like we've accidentally copied that in a few places.